app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+                offsetY = 0.dp,
+                blurRadius = 4.dp
+            ),
+    ) {
+        Box (
+            modifier = Modifier
+                .width(180.dp)
+                .height(270.dp)
+                .padding(vertical = 2.dp)
+                .shadow(
+                    color = gray500,
+                    offsetX = 0.dp,
+                    offsetY = 0.dp,
+                    blurRadius = 4.dp
+                )
+                .clickable {
+                    if (movie != null) {
+                        navController.navigate(Navigator.MovieDetailsPage(title = movie.title, id = movie.id))
+                    }
+                    if (show != null) {
+                        navController.navigate(Navigator.TvDetailsPage(show = show.name, id = show.id))
+                    }
+                },
+            contentAlignment = Alignment.TopStart
+        ) {
+            Column {
+                if (movie != null) {
+                    Box(modifier = Modifier) {
+                        ImageAsync(
+                            contentDescription = "${ catagory } ${ movie.title }",
+                            clip = false,
+                            backDropPath = movie.poster_path
+                        )
+                        Ribbon(paddingStart = 2.dp)
+                    }
+                }
+                if (show != null) {
+                    Box(modifier = Modifier) {
+                        ImageAsync(
+                            contentDescription = "${ catagory } ${ show.name }",
+                            clip = false,
+                            backDropPath = show.poster_path
+                        )
+                        Ribbon(paddingStart = 2.dp)
+                    }
+                }
+            }
+        }
+        if (movie != null) {
+            Details(catagory = catagory, rank = rank, movie = movie, show = null, person = null, showCircleI = true)
+        }
+        if (show != null) {
+            Details(catagory = catagory, rank = rank, show = show, movie = null, person = null, showCircleI = true)
+        }
+    }
+}
+
+@Composable
+fun PosterBoxB(
+    catagory: String,
+    movie: MovieResult,
+    navController: NavController
+) {
+    Column (modifier = Modifier.height(574.dp)) {
+        Row {
+            // upcoming date
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 18.dp, top = 4.dp, bottom = 12.dp),
+                contentAlignment = Alignment.CenterStart
+            ) {
+                //"2024-07-31"
+                val date = movie.release_date.split("-")
+                val monthMap = mapOf("01" to "Jan", "02" to "Feb", "03" to "Mar", "04" to "Apr", "05" to "May", "06" to "Jun", "07" to "Jul", "08" to "Aug", "09" to "Sep", "10" to "Oct", "11" to "Nov", "12" to "Dec",)
+
+                Text(
+                    text = "${ monthMap[date[1]] }-${ date[2] }",
```

Comment:
Instead of parsing the date here, this should probably happen at the deserialization stage when you get back the response. Instead of storing the release date as a string, parse it into a Date object at that time.

https://stackoverflow.com/questions/18473011/retrofit-gson-serialize-date-from-json-string-into-java-util-date